sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -288,6 +288,9 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
    * </li>
    * <li>`allowNumericLeadingZeros` (default `false`): allows leading zeros in numbers
    * (e.g. 00012)</li>
+   * <li>`mode` (default `PERMISSIVE`): allows a mode for dealing with corrupt records
+   * during parsing. When fails to parse, `PERMISSIVE` mode sets `null`, `DROPMALFORMED` drops the
+   * record and `FAILFAST` throws an exception.<li>
```

Comment:
I think we need to say more about these 3 modes. From the tests, it looks to me that:

* `PERMISSIVE` mode will set other fields to null when meet a corrupted record, and put the malformed string into a new field configured by `spark.sql.columnNameOfCorruptRecord`.
* `DROPMALFORMED` mode will ignore corrupted records and append a new field which is always null to the output.
* `FAILFAST` mode will throw an exception.

It will be better if you can expand this doc and add some examples.